### PR TITLE
LPS-138417 - Keycloak tests

### PR DESCRIPTION
**Pull request by weller0246**
https://issues.liferay.com/browse/LPS-138417

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/openidconnect/OpenIDConnect.testcase [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+		SystemSettings.gotoConfiguration(
+			configurationCategory = "SSO",
+			configurationName = "OpenID Connect",
+			configurationScope = "Virtual Instance Scope");
+
+		AssertNotVisible(locator1 = "OpenIDConnect#TOKEN_REFRESH_OFFSET");
+	}
+
+	@description = "This is a use cae for LPS-124900 TC-4: Verify that user session is terminated after a new access token is generated when the Refresh Token Offset is higher. "
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/openidconnect/OpenIDConnect.testcase:30]: cae -> case

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/openidconnect/OpenIDConnect.testcase [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+	@priority = "5"
+	test AssertUserIsLoggedOutWithHigherRefreshToken {
+		property keycloak.enabled = "true";
+		property osgi.module.configuration.file.names = "com.liferay.portal.security.sso.openid.connect.internal.configuration.OpenIdConnectProviderConfiguration~default.config:com.liferay.portal.security.sso.openid.connect.configuration.OpenIdConnectConfiguration.config";
+		property osgi.module.configurations = ":enabled=B&quot;true&quot;${line.separator}tokenRefreshOffset=I&quot;30&quot;";
+		property portal.upstream = "marcell";
+		property test.name.skip.portal.instance = "OpenIDConnect#AssertUserIsStillLoggedInAfterAccessTokenExpiresAndRenew";
+
+		OpenIDConnect.copyOSGIConnectionConfiguration(OSGiConfigFileName = "com.liferay.portal.security.sso.openid.connect.internal.configuration.OpenIdConnectProviderConfiguration_default.config");
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/openidconnect/OpenIDConnect.testcase:39]: _default -> ~default

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/openidconnect/OpenIDConnect.testcase [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+			configurationName = "OpenID Connect",
+			configurationScope = "System Scope");
+
+		AssertTextEquals.assertPartialText(
+			locator1 = "Message#INFO",
+			value1 = "This configuration is not saved yet. The values shown are the default.");
+
+		Type(
+			locator1 = "OpenIDConnect#TOKEN_REFRESH_OFFSET",
+			value1 = "60");
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/openidconnect/OpenIDConnect.testcase:95]: Test with a different value than the default or verify the default is present before typing.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/openidconnect/OpenIDConnect.testcase [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+		AssertClick(
+			locator1 = "Button#SUBMIT",
+			value1 = "Save");
+
+		AssertTextEquals(
+			locator1 = "OpenIDConnect#TOKEN_REFRESH_OFFSET",
+			value1 = "60");
+	}
+
+	@description = "This is a use case for LPS-124898 TC-1: Verify Token Refresh Offset cannot be saved with a value under 30 second.Blocked by LPS-138193"
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/openidconnect/OpenIDConnect.testcase:106]: second.Blocked -> second. Blocked

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/openidconnect/OpenIDConnect.testcase [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+	test VerifyTokenRefreshOffsetMinimum {
+		property portal.upstream = "marcell";
+		property portal.upstream = "quarantine";
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/openidconnect/OpenIDConnect.testcase:109]: No need for quarantine if the test works as expected

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/openidconnect/OpenIDConnect.testcase [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+			value1 = "Save");
+
+		AssertConsoleTextPresent(value1 = "Token refresh offset needs to be at least 30 seconds");
+
+		AssertTextEquals(
+			locator1 = "OpenIDConnect#TOKEN_REFRESH_OFFSET",
+			value1 = "30");
+	}
+
+	@description = "This is a use cae for LPS-124900 TC-2: Verify that user session is not terminated after a new access token is generated when Refresh Token Offset is not higher. "
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/openidconnect/OpenIDConnect.testcase:141]: cae -> case

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/openidconnect/OpenIDConnect.testcase [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+	@priority = "5"
+	test VerifyUserIsStillLoggedInAfterAccessTokenExpiresAndRenew {
+		property keycloak.enabled = "true";
+		property osgi.module.configuration.file.names = "com.liferay.portal.security.sso.openid.connect.configuration.OpenIdConnectConfiguration.config";
+		property portal.upstream = "marcell";
+		property test.name.skip.portal.instance = "OpenIDConnect#AssertUserIsStillLoggedInAfterAccessTokenExpiresAndRenew";
+
+		OpenIDConnect.copyOSGIConnectionConfiguration(OSGiConfigFileName = "com.liferay.portal.security.sso.openid.connect.internal.configuration.OpenIdConnectProviderConfiguration_default.config");
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/openidconnect/OpenIDConnect.testcase:149]: _default -> ~default

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/openidconnect/OpenIDConnect.testcase [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+
+	@description = "This is a use case for LPS-124898 TC-3: Verify that user session is not terminated after a new access token is generated in a clustered environment "
+	test VerifyUserIsStillLoggedInAfterAccessTokenExpiresAndRenewInClusteredEnvironment {
+		property keycloak.enabled = "true";
+		property osgi.module.configuration.file.names = "com.liferay.portal.security.sso.openid.connect.internal.configuration.OpenIdConnectProviderConfiguration~default.config:com.liferay.portal.security.sso.openid.connect.configuration.OpenIdConnectConfiguration.config";
+		property osgi.module.configurations = ":enabled=B&quot;true&quot;${line.separator}tokenRefreshOffset=I&quot;30&quot;";
+		property portal.upstream = "marcell";
+		property test.name.skip.portal.instance = "OpenIDConnect#TokenRefreshOffsetCanBeSetAtSystemLevel";
+
+		OpenIDConnect.copyOSGIConnectionConfiguration(OSGiConfigFileName = "com.liferay.portal.security.sso.openid.connect.internal.configuration.OpenIdConnectProviderConfiguration_default.config");
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/openidconnect/OpenIDConnect.testcase:172]: ~default

**Comment by weller0246** [comment]
ci:test:security:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:sf

**Comment by weller0246** [comment]
ci:test:security:nocompile

**Pull request closed without merging**